ssl/ssl_cert.cc

```diff
@@ -300,49 +310,55 @@ static int cert_set_chain_and_key(
     }
   }
 
-  cert->privatekey = UpRef(privkey);
+  int idx = cert->cert_privatekey_idx;
```

Comment:
What sets this index value to the correct position?

Reply:
Great question, that's part of the broader functionalities being introduced in the next PR (when we actually allow different cert types to use different slots). As of now, `cert_privatekey_idx` can only be `SSL_PKEY_RSA` and is never changed.
Also please let me know if we should come up with a better variable name other than `cert_privatekey_idx` :)